Add diamond nodes

planger opened the pull request:
This change introduces a diamond shaped node with proper anchoring of connections. With that, we address usecase 2 of #235. See in particular https://github.com/theia-ide/sprotty/issues/235#issuecomment-419215072.

If you want to try it out, I integrated the new diamond node into the circles example in branch https://github.com/planger:sprotty/tree/issue235-2-with-example. Anyway, below are two screenshots of a couple of nodes with equal and different width/height.

![image](https://user-images.githubusercontent.com/588090/45445089-976d3b00-b6c9-11e8-815e-6699579c99cc.png)

![image](https://user-images.githubusercontent.com/588090/45445376-545f9780-b6ca-11e8-983b-021208703d09.png)

planger committed:
Add diamond nodes

This change introduces a diamond shaped node with proper anchoring of connections. With that, we address usecase 2 of #235.

spoenemann reviewed (commented):
Thank you! Note, however, that Sprotty is currently in the process of moving to Eclipse. [We already created the initial contribution for the new Eclipse repository](https://dev.eclipse.org/ipzilla/show_bug.cgi?id=17255), so we would need to reapply your changes on that repository when it's available.

planger committed:
Simplify point shifting for computing the offset of diamond anchors

This change also improves the documentation of types and functions in geometry.ts.

planger commented:
Thanks for your fast feedback! I've simplified the point shifting and improved documentation.

> Note, however, that Sprotty is currently in the process of moving to Eclipse. We already created the initial contribution for the new Eclipse repository, so we would need to reapply your changes on that repository when it's available.

I see. Will the code also be moved to an Eclipse repository with Gerrit, or will it stay on Github? I guess it's easiest to just keep this PR open until the CQ is sorted out? Thanks a lot!

JanKoehnlein commented:
We're going to stick to GitHub.

spoenemann commented:
@planger I will notify you when the Eclipse GitHub repo is ready. Maybe then you could create a new PR on that repo to make sure the IP requirements of Eclipse are met.

planger commented:
Thanks a lot! I'll be happy to create a new PR once the new Eclipse repo is available.

spoenemann commented:
@planger the new repo is ready for contributions: https://github.com/eclipse/sprotty

spoenemann commented:
@planger the new repo is ready for contributions: https://github.com/eclipse/sprotty

spoenemann commented:
@planger the new repo is ready for contributions: https://github.com/eclipse/sprotty

spoenemann commented:
@planger the new repo is ready for contributions: https://github.com/eclipse/sprotty

spoenemann commented:
@planger the new repo is ready for contributions: https://github.com/eclipse/sprotty

spoenemann commented:
@planger the new repo is ready for contributions: https://github.com/eclipse/sprotty

spoenemann commented:
@planger the new repo is ready for contributions: https://github.com/eclipse/sprotty

spoenemann commented:
@planger the new repo is ready for contributions: https://github.com/eclipse/sprotty

planger commented:
@spoenemann Great, thanks for letting me know. I made myself a task to contribute this change in the new Eclipse repo. I'll do that at latest beginning of next week. Thanks!

planger commented:
@spoenemann Thanks again for pointing me to the new repo. I've opened a new PR for this change, see https://github.com/eclipse/sprotty/pull/3.
